Base/QTCLI/vtkMRMLScalarVectorDWIVolumeResampler.cxx

```diff
@@ -0,0 +1,179 @@
+/*==============================================================================
+
+  Program: 3D Slicer
+
+  Copyright(c) Kitware Inc.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+==============================================================================*/
+
+#include "vtkMRMLScalarVectorDWIVolumeResampler.h"
+
+// Slicer includes
+#include "qSlicerCLIModule.h"
+#include "qSlicerCoreApplication.h"
+#include "qSlicerModuleManager.h"
+#include "vtkSlicerCLIModuleLogic.h"
+
+// MRML includes
+#include "vtkMRMLAbstractVolumeResampler.h"
+#include "vtkMRMLCommandLineModuleNode.h"
+#include "vtkMRMLTransformNode.h"
+#include "vtkMRMLVolumeNode.h"
+
+//----------------------------------------------------------------------------
+vtkStandardNewMacro(vtkMRMLScalarVectorDWIVolumeResampler);
+
+//----------------------------------------------------------------------------
+void vtkMRMLScalarVectorDWIVolumeResampler::PrintSelf(ostream& os, vtkIndent indent)
+{
+  this->Superclass::PrintSelf(os, indent);
+}
+
+//----------------------------------------------------------------------------
+bool vtkMRMLScalarVectorDWIVolumeResampler::Resample(vtkMRMLVolumeNode* inputVolume,
+                                                     vtkMRMLVolumeNode* outputVolume,
+                                                     vtkMRMLTransformNode* resamplingTransform,
+                                                     vtkMRMLVolumeNode* referenceVolume,
+                                                     int interpolationType,
+                                                     int windowedSincFunction,
+                                                     const ResamplingParameters& resamplingParameter)
+{
+  // A helper RAII class to ensure a vtkMRMLNode is removed from its scene
+  // when the enclosing scope is exited, regardless of how it is exited.
+  struct vtkMRMLNodeCleanup
+  {
+  public:
+    vtkMRMLNodeCleanup(vtkMRMLScene* scene, vtkMRMLNode* node)
+      : Scene(scene), Node(node)
+    {
+    }
+    ~vtkMRMLNodeCleanup()
+    {
+      if (this->Scene != nullptr)
```

Comment:
Also check that `this->Node != nullptr` to avoid ` vtkDebugMacro("RemoveNode: unable to remove null node");`.  Alternatively, instantiate `vtkMRMLNodeCleanup` after line 97 instead of on line 92.